journald/lib_journald.hpp

```diff
@@ -18,10 +18,105 @@ namespace journald {
+// prior to executors can override the global values.
+struct LoggerFlags : public virtual flags::FlagsBase
+{
+  LoggerFlags()
+  {
+    add(&LoggerFlags::destination_type,
+        "destination_type",
+        "Determines where logs should be piped.\n"
+        "Valid destinations include: 'journald', 'sandbox', or 'both'.",
+        "journald",
+        [](const std::string& value) -> Option<Error> {
+          if (value != "journald" && value != "sandbox" && value != "both") {
+            return Error("Invalid destination type: " + value);
+          }
+
+          return None();
+        });
+
+    add(&LoggerFlags::max_stdout_size,
+        "max_stdout_size",
+        "Maximum size, in bytes, of a single stdout log file.\n"
+        "Defaults to 10 MB.  Must be at least 1 (memory) page.",
+        Megabytes(10),
+        &LoggerFlags::validateSize);
+
+    add(&LoggerFlags::logrotate_stdout_options,
+        "logrotate_stdout_options",
+        "Additional config options to pass into 'logrotate' for stdout.\n"
+        "This string will be inserted into a 'logrotate' configuration file.\n"
+        "i.e.\n"
+        "  /path/to/stdout {\n"
+        "    <logrotate_stdout_options>\n"
+        "    size <max_stdout_size>\n"
+        "  }\n"
+        "NOTE: The 'size' option will be overriden by this module.");
+
+    add(&LoggerFlags::max_stderr_size,
+        "max_stderr_size",
+        "Maximum size, in bytes, of a single stderr log file.\n"
+        "Defaults to 10 MB.  Must be at least 1 (memory) page.",
+        Megabytes(10),
+        &LoggerFlags::validateSize);
+
+    add(&LoggerFlags::logrotate_stderr_options,
+        "logrotate_stderr_options",
+        "Additional config options to pass into 'logrotate' for stderr.\n"
+        "This string will be inserted into a 'logrotate' configuration file.\n"
+        "i.e.\n"
+        "  /path/to/stderr {\n"
+        "    <logrotate_stderr_options>\n"
+        "    size <max_stderr_size>\n"
+        "  }\n"
+        "NOTE: The 'size' option will be overriden by this module.");
+  }
+
+  static Option<Error> validateSize(const Bytes& value)
+  {
+    if (value.bytes() < os::pagesize()) {
+      return Error(
+          "Expected --max_stdout_size and --max_stderr_size of "
+          "at least " + stringify(os::pagesize()) + " bytes");
+    }
+
+    return None();
+  }
+
+  std::string destination_type;
+
+  Bytes max_stdout_size;
+  Option<std::string> logrotate_stdout_options;
+
+  Bytes max_stderr_size;
+  Option<std::string> logrotate_stderr_options;
+};
+
+
+struct Flags : public virtual LoggerFlags
 {
   Flags()
   {
+    add(&Flags::environment_variable_prefix,
+        "environment_variable_prefix",
+        "Prefix for environment variables meant to modify the behavior of\n"
+        "the logrotate logger for the specific executor being launched.\n"
+        "The logger will look for four prefixed environment variables in the\n"
```

Comment:
Definitely don't want executors changing the `companion_dir` (which will just cause their executor to fail).  

`libprocess_num_worker_threads` is more of an optimization that is more important for the operator (who might be concerned about running out of threads) than the executor, as the executor should be isolated from other processes on the node.